datasets/flwr_datasets/visualization/label_distribution.py

```diff
@@ -0,0 +1,643 @@
+    >>     },
+    >> )
+    >> partitioner = fds.partitioners["train"]
+    >> axis, dataframe = plot_label_distributions(
+    >>     partitioner=partitioner,
+    >>     label_name="label"
+    >> )
+
+    Alternatively you can visualize each partition in terms of fraction of the data
+    available on that partition instead of the absolute count
+
+    >> from flwr_datasets import FederatedDataset
+    >> from flwr_datasets.partitioner import DirichletPartitioner
+    >> from flwr_datasets.visualization import compare_label_distribution
+    >>
+    >> fds = FederatedDataset(
+    >>     dataset="cifar10",
+    >>     partitioners={
+    >>         "train": DirichletPartitioner(
+    >>             num_partitions=20,
+    >>             partition_by="label",
+    >>             alpha=0.3,
+    >>             min_partition_size=0,
+    >>         ),
+    >>     },
+    >> )
+    >> partitioner = fds.partitioners["train"]
+    >> axis, dataframe = plot_label_distributions(
+    >>     partitioner=partitioner,
+    >>     label_name="label"
+    >>     size_unit="percent",
+    >> )
+    >>
+
+    You can also visualize the data as a heatmap by changing the `plot_type` from
+    default "bar" to "heatmap"
+
+    >> from flwr_datasets import FederatedDataset
+    >> from flwr_datasets.partitioner import DirichletPartitioner
+    >> from flwr_datasets.visualization import compare_label_distribution
+    >>
+    >> fds = FederatedDataset(
+    >>     dataset="cifar10",
+    >>     partitioners={
+    >>         "train": DirichletPartitioner(
+    >>             num_partitions=20,
+    >>             partition_by="label",
+    >>             alpha=0.3,
+    >>             min_partition_size=0,
+    >>         ),
+    >>     },
+    >> )
+    >> partitioner = fds.partitioners["train"]
+    >> axis, dataframe = plot_label_distributions(
+    >>     partitioner=partitioner,
+    >>     label_name="label"
+    >>     size_unit="percent",
+    >>     plot_type="heatmap",
+    >>     annot=True,
+    >> )
+
+    You can also visualize the returned DataFrame in Jupyter Notebook
+    >> df.style.background_gradient(axis=None)
+    """
+    _validate_parameters(plot_type, size_unit, partition_id_axis)
+
+    if label_name not in partitioner.dataset.column_names:
+        raise ValueError(
+            f"The specified 'label_name': '{label_name}' is not present in the "
+            f"dataset."
+        )
+
+    if max_num_partitions is None:
+        max_num_partitions = partitioner.num_partitions
+    else:
+        max_num_partitions = min(max_num_partitions, partitioner.num_partitions)
+    assert isinstance(max_num_partitions, int)
+    partitions = [partitioner.load_partition(i) for i in range(max_num_partitions)]
+
+    partition = partitions[0]
+    try:
+        unique_labels = partition.features[label_name].str2int(
+            partition.features[label_name].names
+        )
+    except AttributeError:  # If the label_name is not formally a Label
```

Comment:
what's an example of dataset and `label_name` combination that would trigger this?

Reply:
For example `adult-census-income` dataset https://huggingface.co/datasets/scikit-learn/adult-census-income with "income" with serves as label because it's of type Value(sth) not ClassLabel. (a little bit of background info) each column in HF dataset has type. You specify Features (https://huggingface.co/docs/datasets/v2.19.0/en/package_reference/main_classes#datasets.Features) and if not treated specially the column is just a Value. However, you can properly make a column as a label. If you do that then the values are represented as ints and only then can you transform them to full string names by using features[label_name].int2str and do the reverse `str2int`. The str2int does not exist for a column of type Value but only for type ClassLabel

Reply:
ok! i get it. thanks for the explanation 🙏